rust/lance/src/dataset.rs

```diff
@@ -1657,6 +1691,69 @@ impl Dataset {
         Ok(fragments)
     }
 
+    /// Modify columns in the dataset, changing their name, type, or nullability.
+    ///
+    /// If a column has an index, it's index will be preserved or transformed to
+    /// the new type as part of the operation.
```

Comment:
Index transformation is only future right?  Once we support data type changes?

Reply:
Yeah I should remove that for now.